nemo/collections/asr/parts/mixins/transcription.py

```diff
@@ -69,13 +69,18 @@ class TranscribeConfig:
 def move_to_device(batch, device, non_blocking=False):
     """
     Recursively move all tensors in `batch` to `device`.
+    Supports tensors, lists, tuples, dictionaries, and dataclasses.
     """
     if isinstance(batch, torch.Tensor):
         return batch.to(device, non_blocking=non_blocking)
     elif isinstance(batch, (list, tuple)):
-        return [move_to_device(x, device, non_blocking) for x in batch]
+        return type(batch)(move_to_device(x, device, non_blocking) for x in batch)
```

Comment:
FYI @titu1994 , returned tuples would always become lists. And namedtuples would become lists as well. This changes behavior, but seems much more preferable.